server/routes/api/messages.js

```diff
@@ -43,4 +51,88 @@ router.post("/", async (req, res, next) => {
   }
 });
 
+// Update mssage read status in conversation
+router.patch("/", async (req, res, next) => {
+  try {
+    if (!req.user) {
+      return res.sendStatus(401);
+    }
+
+    const { senderId, activeConversationId } = req.body;
+
+    // check if activeConversationId is exist
+    if (activeConversationId) {
+      let conversation = await Conversation.findConversationById(
+        activeConversationId,
+        senderId,
+        req.user.id
+      );
+
+      // if not exist return error
+      if (!conversation) return res.json(conversation);
+
+      // if activeConversationId exists, start update messages
+      const messages = await Message.findAll({
+        where: {
+          conversationId: activeConversationId,
+          senderId: {
+            [Op.eq]: senderId,
+          },
+          readStatus: {
+            [Op.is]: false,
+          },
+        },
+      });
+
+      // there is no messages, then return empty array
+      if (!Array.isArray(messages) || !messages.length)
+        return res.json({ messages });
+
+      // if have messages, update read message status
+      messages.forEach((message) => {
+        if (!message.readStatus) message.readStatus = true;
+        message.readById = senderId;
+        message.readTime = Date.now();
+      });
+
+      // save updated messages
+      await Promise.all(messages.map(async (message) => await message.save()));
+      return res.json({ messages });
+    }
```

Comment:
This logic can be simplified significantly. You can update the message is one call with `Message.update(...)`

I do like that you used Promise.all() 🙂  👍 

And are you sure you need to be returning the messages back to the client? They can already mark those messages as read on the client so simply returning a `204` and no data would be sufficient.